### ISSUE TITLE
During install 'make -j4 check' fails with Thrift errors

### ISSUE THREAD

**Issue by jwbensley** [p4lang/p4c#773]
Hi guys

I'm trying to compile on an Ubuntu 16 virtual machine but `make -j4 check` shows that all the BMv2 tests fail (all others are fine, see attached `make-check.log` file).

I have attached the `test-suite.log` file showing the BMv2 failed test results. They are all relating to the Thrift module.

# System info
```
bensley@ubuntu-netops:~/P4$ uname -a
Linux ubuntu-netops 4.4.0-83-generic #106-Ubuntu SMP Mon Jun 26 17:54:43 UTC 2017 x86_64 x86_64 x86_64 GNU/Linux

bensley@ubuntu-netops:~/P4$ lsb_release -a
No LSB modules are available.
Distributor ID:	Ubuntu
Description:	Ubuntu 16.04.2 LTS
Release:	16.04
Codename:	xenial

bensley@ubuntu-netops:~/P4$ pip freeze | grep -iE "nnpy|thrift"
nnpy==1.4.1
thrift==0.9.2
thriftpy==0.3.9
```

# Installation Steps

I have checked out this reop: https://github.com/p4lang/behavioral-model

I then ran the `install_deps.sh` script, Thrift, Nanomsg and nnpy are all installed. I have then compiled and installed the contents of that repo without issue.

I have checked out this repo (https://github.com/p4lang/p4c) and installed all the Ubuntu 16.04 dependencies, even the optional documentation dependencies. I have compiled and installed Google proto bufs from source (latest version) which installed (and all it's dependencies) without issue.

So far everything has compiled and installed fine. When I tried to compile P4 `make -j` runs fine and finishes with no errors however `make -j4 check` lists errors with all the BMv2 tests. I want to compile to x86 to run tests on a Linux box so that is quite important to me.

All 200-ish errors are basically the same, a class is missing from Thrift:

```
  File "/usr/local/lib/python2.7/dist-packages/bm_runtime/standard/Standard.py", line 9, in <module>
    from thrift.Thrift import TType, TMessageType, TFrozenDict, TException, TApplicationException
ImportError: cannot import name TFrozenDict
```

I'm not sure what to do here (should I ignore these errors or will there be a problem further down the road)?

In the `make-log.log` file I have attached I see `  CXX      backends/bmv2/bmv2.o` so it looks like BMv2 compiles without issue.

[logs.zip](https://github.com/p4lang/p4c/files/1144666/logs.zip)

**Comment by antoninbas** [comment]
This usually means that your Thrift compiler version and your Thrift Python package version are not the same. Try running `thrift -v` and see if it matches -should be `0.9.2` based on the information you gave, but I'm pretty sure it will say `0.9.3`. If I'm right just upgrade the Python package to `0.9.3` with `pip`.

**Comment by jwbensley** [comment]
@antoninbas Thanks for the info.

I was using an "old" virtual machine by which I mean, I'm not sure what state it was left in from when I last used it. I set up a new VM and everything is compiling fine now. It might be worth adding to the README about _only_ using the `install_deps.sh` for the https://github.com/p4lang/behavioral-model repo, I actually couldn't run `make -j4 check` and pass all tests without using that install script, I had installed the latest version of each requirement and that didn't seem to work. I had to use those install scripts with `-f` and use the exact versions references they reference.
